src/components/detail/Detail.js

```diff
@@ -11,6 +12,31 @@ const Detail = (props) => {
   const { user } = useAuthContext();
   // console.log(user);
 
+
+  // Popup Function
+  const [popUp, setPopUp] = useState(false)
+
+  const popUpRef = useRef()
+
+  useEffect(() => {
+    
+    const handle = (e) => {
+
+      // If popup exist and clicked outside of the popup
+      if (popUpRef.current && !popUpRef.current.contains(e.target)) {
+
+        // Set popup state to false (close it)
+        setPopUp(false)
+      }
+    }
+    
+    document.addEventListener("mousedown", handle)
+
+    // Unmount the eventlistner
+    return () => document.removeEventListener("mousedown", handle)
+  })

```

Comment:
Sorry for asking many times, but can you add second argument like [] to useEffect basically?